graphistry/umap_utils.py

```diff
@@ -119,12 +167,18 @@ def umap_lazy_init(
         negative_sample_rate=5,
         n_components: int = 2,
         metric: str = "euclidean",
+        engine: str = "auto", #this seems to override user input
+        # has_umap: str = "True",
     ):
 
         # FIXME remove as set_new_kwargs will always replace?
-        has_umap, _, umap = lazy_umap_import_has_dependancy()
+        engine_resolved = resolve_umap_engine(engine)
+        if engine_resolved == "umap_learn":
```

Comment:
cleaner like

```python
if engine_resolved == "umap_learn":
    _, _, umap_engine = ...
elif engine_resolved == "umap":
   _, _, umap_engine = ..
else:
    raise ValueError('No umap engine, ensure "auto", "umap", or "cuml", and the library is installed`)
```

b/c later guaranteed to have it and have the obj, and can do any conditionals on engine